### DISCUSSION TITLE
How to get the header in the SignalReceived request header。

### DISCUSSION THREAD

**Discussion by tuohaibei** [elsa-workflows/elsa-core#4569] [Q&A] [answered]
Hi 
I want to access request token from SignalReceived ,because i want to permission verification.
how i could get that,i know how get access token from http endpoint.but i don't know how get header from SignalReceived 
@sfmskywalker 

![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/0c72a7d6-4ced-4671-8a1a-d364523d4616)

![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/0b55a50b-af79-4609-805a-bc677950aded)

**Comment by tuohaibei** [comment]
Looking forward to your suggestions @sfmskywalker

**Reply by tuohaibei** [reply, marked as answer]
I have fixed the problem
SignalReceived is essentially an http request,so i get the header through IHttpContextAccessor,so i inject it in custom activity and get the header.

**Reply by sfmskywalker** [reply]
Hi @tuohaibei , I'm happy to see that you figured it out!
